diesel/src/pg/query_builder/distinct_on.rs

```diff
@@ -33,16 +34,18 @@ where
     }
 }
 
-impl<ST, F, S, D, W, O, LOf, G, Selection> DistinctOnDsl<Selection>
-    for SelectStatement<F, S, D, W, O, LOf, G>
+impl<ST, F, S, D, W, O, LOf, G, H, Selection> DistinctOnDsl<Selection>
```

Comment:
The order here seems different than `SelectStatement` which is why you had to write `NoLockingClause` all the time below. You don't need to do that if you change the `SelectStatement`.

Reply:
Fixed.